Modules/AbRouter/Services/Events/Stats/EventStats.php

```diff
@@ -0,0 +1,158 @@
+             */
+
+            $relatedUsersIds = $event
+                ->relatedUsers
+                ->reduce(function (array $acc, RelatedUser $relatedUser) {
+                    if (empty($relatedUser->related_user_id)) {
+                        return $acc;
+                    }
+
+                    $acc[] = $relatedUser->related_user_id;
+                    return $acc;
+                }, []);
+
+            $userEventKey = $event->event . '_' . $event->user_id;
+
+            if (!empty($event->user_id) && isset($userEventAdded[$userEventKey])) {
+                continue;
+            }
+
+            sort($relatedUsersIds);
+            if (!empty($relatedUsersIds)) {
+                $relatedUsersKey = $event->event . '_' . join('_', $relatedUsersIds);
+            } else {
+                $relatedUsersKey = '';
+            }
+
+            if (!empty($relatedUsersKey) && isset($userEventAdded[$relatedUsersKey])) {
+                continue;
+            }
+
+            $hasUserInRelated = false;
+            foreach ($relatedUsersIds as $relatedUsersId) {
+                if (isset($uniqUsers[$relatedUsersId])) {
+                    $hasUserInRelated = true;
+                    break;
+                }
+            }
+
+            if (!$hasUserInRelated && !isset($uniqUsers[$event->user_id])) {
+                continue;
+            }
+
+            if ($hasUserInRelated) {
+                $userEventAdded[$relatedUsersKey] = true;
+            }
+            if (isset($uniqUsers[$event->user_id])) {
+                $userEventAdded[$userEventKey] = true;
+            }
+
+            if ($enableDateCounter) {
+                $convertDate = $event->created_at->format('Y-m-d');
+
+                if (!isset($eventCounters[$event->event][$convertDate])) {
+                    $eventCounters[$event->event][$convertDate] = 0;
+                }
+
+                $eventCounters[$event->event][$convertDate] ++;
+
+                continue;
+            }
+
+            if (!isset($eventCounters[$event->event])) {
+                $eventCounters[$event->event] = 0;
+            }
+
+            $eventCounters[$event->event] ++;
+
+            $eventType = !empty($allDisplayEvents)
+                ? array_reduce($allDisplayEvents,
+                    function (array $acc, $allDisplayEvents) use ($event) {
+                        if ($allDisplayEvents['event_name'] === $event->event) {
+                           $acc['type'] = $allDisplayEvents['type'];
+                           return $acc;
+                        }
+                        return $acc;
+                    }, [])
+                : ['type' => ''];
+
+            if ($eventType['type'] === 'summarizable') {
+                if (!isset($eventCounters['summarization'][$event->event])) {
+                    is_numeric($event->value) ?
+                        $eventCounters['summarization'][$event->event] = $event->value

```

Comment:
Please, put value of $eventCounters['summarization'][$event->event] to $eventCounters['summarization']